### ISSUE TITLE
Incorporate principles

### ISSUE THREAD

**Issue by iluwatar** [iluwatar/java-design-patterns#425]
I've been helping out a friend with [this fork](https://github.com/iluwatar/programming-principles) but lately I've gained a vision how the principles should be presented.

I suggest that we incorporate the principles to the java-design-patterns.com web site, in their own section. The presentation of the principles would be similar to [patterns page](http://java-design-patterns.com/patterns/). The principles would have categories, tags, search function and all the goodies that we have for the patterns. I'm not sure if they would need code samples, but I guess technically this would be possible too.

@npathai @markusmo3 what do you think?

**Comment by npathai** [comment]
@iluwatar We had a discussion on this in #208 and decided to just link `web pro` link on our website. Can we not help the principles website on http://webpro.github.io/programming-principles/ to convert to our style?

**Comment by markusmo3** [comment]
@iluwatar Looks nice, and incorporating it into the website would be awesome.
For it to work, the website just needs the *.md files and a new template, so no biggie there.

I think we need to start discussing about the future plans of this project, if we decide to add the principles:
* Currently we have one master and one website, purely about JDP. If we incorporate the principles, we'd have inconsistent naming. So it would have to be organised somehow (pull website out into separate repo / adding principles to our repo, maybe branch, maybe something else...)
* As mentioned, a new name for the website would be nice, so others can clearly see that the website isnt just about patterns, but also principles.
* As we grow bigger and bigger, especially if the principles are incorporated, we should start thinking about using an Organization. And with that comes the question of adding more collaborators, like webpro himself, if he wants. One big pro of using an Org. would be that we could use a [Org. site](https://help.github.com/articles/user-organization-and-project-pages/) and have the sources/*.md separate from the website code.

@npathai helping them convert would be an option, if the website wouldnt be so complicated and intertwined with travis and other systems. I'm planning on maybe composing a better guide to how our website works. Overall, adding the jekyll stuff on our end only to show PP, is much easier than trying to give them/him our backend.

TL;DR :+1: for adding them

**Comment by iluwatar** [comment]
Yeah, I remember the discussion @npathai and so far it has served us well that way. I would just like to make the principles first class citizen in this project. They are kind of natural fit, principles and patterns. Of course, this would need to be agreed with webpro but I don't see a problem there. But should we expand this way, I would like to do this unanimously.

@markusmo3 here's my comments:
- We could think of another name. Programming, principles, patterns...
- Converting to an organization is certainly possible although we need to study what that means and how it affects the current way we work.

**Comment by npathai** [comment]
@iluwatar @markusmo3 Yes even I am up for incorporating them on our website. Also we need new naming. I will think of some options. Organization would surely be next step towards expanding our reach. :+1: For that. Why not incorporate webpro in this discussion as well.

**Comment by iluwatar** [comment]
I've just sent the proposal.

**Comment by markusmo3** [comment]
resolved as of #481
